debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/gc/GCObjectHeapIteratorAddressOrderedList_V1.java

```diff
@@ -133,12 +140,37 @@ public int compare(U8Pointer[] o1, U8Pointer[] o2)
 					return o1[0].compare(o2[0]);
 				}
 			});
-		excludedRanges = new U8Pointer[excludedRangeList.size()][2];
+		excludedRanges = new U8Pointer[excludedRangeList.size()][];
 		excludedRangeList.toArray(excludedRanges);
 		currentExcludedRange = 0;
 	}
-	
-	private U8Pointer adjustedToRange(U8Pointer ptr, U8Pointer base, U8Pointer top)
+
+	protected U8Pointer getRealHeapTop(J9ModronThreadLocalHeapPointer threadLocalHeap, U8Pointer heapTop, U8Pointer base, U8Pointer top) throws CorruptDataException
+	{
+		return heapTop;
+	}
+
+	protected U8Pointer getRealHeapAlloc(J9ModronThreadLocalHeapPointer threadLocalHeap, U8Pointer heapAlloc, U8Pointer base, U8Pointer top) throws CorruptDataException
+	{
+		U8Pointer realHeapAlloc = null;
+		try {
+			if (null == method) {
+				method = threadLocalHeap.getClass().getMethod("realHeapAlloc");
+			}
+			realHeapAlloc = adjustedToRange((U8Pointer)method.invoke(threadLocalHeap), base, top);
+		} catch (NoSuchMethodException e1) {
+			throw new CorruptDataException("caused by NoSuchMethodException: " + e1.getMessage());
+		} catch (IllegalAccessException e2) {
+			throw new CorruptDataException("caused by IllegalAccessException: " + e2.getMessage());
+		} catch (IllegalArgumentException e3) {
+			throw new CorruptDataException("caused by IllegalArgumentException: " + e3.getMessage());
+		} catch (InvocationTargetException e4) {
+			throw new CorruptDataException("caused by InvocationTargetException: " + e4.getMessage());
```

Comment:
If `e4.getTargetException()` is a `CorruptDataException` or a `RuntimeException`, that should be thrown instead.